### ISSUE TITLE
Deleting Guest Author should free up display name

### ISSUE THREAD

**Issue by tobek** [Automattic/Co-Authors-Plus#571]
Had a user run into an issue where they deleted a guest author, and then had to create a new one with the same display name. Since the deleted guest author still existed in the trash, creating a new one with the same name created a `Display name conflicts with another guest author display name` error.

Workarounds:

- Manually go to `.../wp-admin/edit.php?post_status=trash&post_type=guest-author` and permanently delete (or restore) the trashed guest author.
- Or, create a new guest author with a different display name (ideally similar, since slug will be used in author page), and, after saving, go back in and change display name. This works I guess because conflict is actually based on the slug, and the slug is generated from display name.

Potential fixes:

- Permanently delete guest authors by default instead of trashing them.
- Or, in the "display name conflict" error message, either offer to delete (or navigate to) existing guest author, or perhaps expose a link to trashed guest authors so user could go in and sort out.

**Comment by sboisvert** [comment]
@smistephen Want to take a look?

**Comment by smistephen** [comment]
Hmmm.

It... works for me?

Here are my steps:
1) Create guest author with display name Stephen Smith, first name Stephen, last name Smith.
2) Delete guest author.
3) Perform step 1 again.

Step 3 did not produce an error, and it is my understanding that it should.

Thought maybe it only bugs out on guest authors with posts attached, so I published a post with the guest author attached, deleted the guest author (selecting "remove byline"), recreated it, everything still worked just fine.

Tried a third time, same as the second except I reassigned the posts on delete instead of removing the byline. Still, I could not make it break. I even attempted mapping it to a user and then deleting, to no avail.

So I checked out how the delete algorithm works, and found that the code is already set up to permadelete guest authors by default, as was suggested:
https://github.com/Automattic/Co-Authors-Plus/blob/master/php/class-coauthors-guest-authors.php#L1317

I'd think that maybe they were using an old version of the plugin, but the commit that introduced permadeletion is six years old: https://github.com/Automattic/Co-Authors-Plus/commit/b4f38540afe9e682332210066715e2a77c0df518

My suggested way forward would be to get more information from the user about the steps they followed to generate the error. Perhaps they followed an user path I didn't think of.

**Comment by sboisvert** [comment]
>My suggested way forward would be to get more information from the user about the steps they followed to generate the error. Perhaps they followed an user path I didn't think of.

Feel free to ask @tobek directly :)

**Comment by smistephen** [comment]
Hi @tobek, would you mind asking this user for more information? I've been unable to replicate their issue :(

**Comment by tobek** [comment]
Huh thanks for looking into it @sboisvert. Yeah I followed the steps on the instance where we ran into this issue and I also did not run into the problem. I was too quick to assume it happened in such a straightforward way! We're running v3.3.0. I am not sure what path she took but there is a guest author in the trash, which I can't see how to do even on purpose:

![image](https://user-images.githubusercontent.com/3081073/43398700-c0f33baa-9400-11e8-800c-7072e3ce3dc8.png)

I tried some permutations, like linking guest author to a WP user, then removing WP user from that site (this is a multi site network), then deleting the WP user from network entirely. Tried some versions of this with the guest author being on posts vs. not, different assignment of posts after deletion, checking the trash at each step, can't reproduce.

I asked the user but she doesn't remember at all what happened.

Clearly whatever happened is an edge case, and there's a chance it's an interaction with another plugin (though we have very few, only major ones are Jetpack and Edit Flow) and if it was interaction with another plugin such that this happened when doing normal stuff, I'd hope/expect to be able to reproduce it.

I can't look into it any more right now, so feel free to close the issue, though my hunch is that there is some weird user flow out there that does this!
